File: jserver-core/src/main/java/com/github/gr3gdev/jserver/http/impl/RemoteAddressImpl.java
package com.github.gr3gdev.jserver.http.impl;

import com.github.gr3gdev.jserver.http.RemoteAddress;

/**
 * RemoteAddressImpl.
 *
 * @author Gregory Tardivel
 */
public class RemoteAddressImpl implements RemoteAddress {

    private final String ip;
    private final String port;

    public RemoteAddressImpl(String remoteAddress) {
        final String[] data = remoteAddress.substring(1).split(":");
        ip = data[0];
        port = data[1];
    }

    @Override
    public String ip() {
        return null;
    }

    @Override
    public String port() {
        return null;
    }
}

File: jserver-core/src/main/kotlin/com/github/gr3gdev/jserver/http/impl/RemoteAddressImpl.kt
package com.github.gr3gdev.jserver.http.impl

import com.github.gr3gdev.jserver.http.RemoteAddress

class RemoteAddressImpl(private val remoteAddress: String) : RemoteAddress {

    private val ip: String
    private val port: String

    init {
        val data = remoteAddress.substring(1).split(":")
        ip = data[0]
        port = data[1]
    }

    override fun ip(): String {
        return ip
    }

    override fun port(): String {
        return port
    }
}